db/config.c

```diff
@@ -765,6 +761,7 @@ static void enable_snapshot(struct dbenv *dbenv) {
         return;
     }
 
+    set_snapshot_impl(gbl_snap_impl); 
```

Comment:
I didn't previously think to do this here because the default implementation was the original implementation, which doesn't have any specific tunables that need to be toggled.

If either modsnap or newsi is default then we need to toggle its tunables.